gitpod-io/gitpod
# [public-api] Bind OIDC HTTP service to Public API

## Description


Attaches the HTTP OIDC endpoints to Public API. This is part of merging the `iam` component with the `public-api-server` component.

Both `iam` and `public-api-server` will be running these endpoints, and traffic from the proxy remains configured to go to `iam`. But this allows us to test the endpoint works by port-forwarding to it, before make the full switch in subsequent PR.

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/15860

## How to test

1. Preview k8s context
2. `kubectl port-forward deployment/public-api-server 9002:9002`
3. 
	```
	curl localhost:9002/oidc/start
	config not found
	```
4. Config not found is expected, it means we can reach the endpoint and it's handled by `public-api-server`. We will use this endpoint in https://github.com/gitpod-io/gitpod/pull/15940


## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all-ci
      Run Leeway with `--dont-test`
      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
